File: 2524-largest-positive-integer-that-exists-with-its-negative/2524-largest-positive-integer-that-exists-with-its-negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        unordered_map<int, int> map;
        int k = -1;
        for (int i = 0; i < nums.size(); ++i) {
            int temp = ~nums[i] + 1;
            if (map.count(temp)) {
                int tmp = (nums[i] ^ (nums[i] >> 31)) - (nums[i] >> 31);
                k = tmp > k ? tmp : k;
            }
            map[nums[i]] = 1;
        }
        return k;
    }
};

File: 2524-largest-positive-integer-that-exists-with-its-negative/2524-largest-positive-integer-that-exists-with-its-negative.java
class Solution {
    public int findMaxK(int[] nums) {
        Map<Integer, Integer> m = new HashMap<>();
        int k = -1;
        for (int i = 0; i < nums.length; ++i) {
            if (m.containsKey(~nums[i] + 1)) {
                int temp = (nums[i] ^(nums[i] >> 31)) - (nums[i] >> 31);
                k = temp > k ? temp : k;
            }
            m.put(nums[i], 1);
        }
        return k;
    }
}